### PR TITLE
Static errors continuation

### PR DESCRIPTION
**Pull request by shimwell**
Following up on PR #2485 which did a lot of the ground work needed to get mypi running.

I didn't want the 1000 lines of changes already done in #2485 to go to waste and I am keen on seeing more python checking (mypi, flake8, black etc) being made use of in openmc so I am keen to finish this mypi PR

the additional changes on this PR compared to #2485 are the movement of settters so they are directly after the properties. This gets rid of the mypi error that says ```name * already defined```

### CODE REVIEW

**Comment by shimwell** [comment]
This PR attracts merge conflicts like a they are magnetic. One just arrived in tallies.py today and there was one yesterday

File: openmc/data/energy_distribution.py [hunk 1]
```diff
@@ -777,8 +777,6 @@ def from_endf(cls, file_obj, params):
         efl, efh = params[0:2]
         return cls(efl, efh, tm)
 
-
-
 class DiscretePhoton(EnergyDistribution):
```
Comment (paulromano) [inline openmc/data/energy_distribution.py:780]: There should be two lines before a class definition per PEP8
```suggestion

class DiscretePhoton(EnergyDistribution):
```

File: openmc/geometry.py [hunk 2]
```diff
@@ -65,20 +70,19 @@ def bounding_box(self) -> np.ndarray:
...
         check_type('merge surfaces', merge_surfaces, bool)
         self._merge_surfaces = merge_surfaces
 
+    @property
+    def surface_precision(self) -> int:
+        return self._surface_precision
+
+    @property
+    def surface_precision(self):
+        return self._surface_precision
```
Comment (paulromano) [inline openmc/geometry.py:84]:
```suggestion
    @property
    def surface_precision(self) -> int:
        return self._surface_precision
```

File: openmc/mgxs/mgxs.py [hunk 3]
```diff
@@ -3482,6 +3482,7 @@ def prompt(self, prompt):
             self._rxn_type = 'prompt-nu-fission'
 
 
+
```
Comment (paulromano) [inline openmc/mgxs/mgxs.py:3485]:
```suggestion
```

**Review by paulromano — CHANGES_REQUESTED** [review]
Thanks for continuing work on this @shimwell  -- there are a few small things that need fixing but should be good to merge after that.

**Review by paulromano — APPROVED** [review]

**Pull request merged**